### PR TITLE
Feat: Simulated Annealing, analysis and refactoring

### PR DESCRIPTION
**Pull request by PersonligaPersson**
- Identified a complex function `simulated_annealing` (Lizard: 16)
- Manually counted a complexity of 15.
- Refactored the function.
- Wrote a report. 
New complexity (manual and Lizard) is 10.
This relates to issue #6

### CODE REVIEW

File: report.md [hunk 1]
```diff
@@ -34,7 +34,7 @@ The eight functions are presented in the table below. Every function has a CCN o
 | interpolation_search(sorted_collection, item) | searches/interpolation_search.py                     | 10          | 6                       | 6                       | 30           |
 | cycle_sort(list)                              | sorts/cycle_sort.py                                  | 10          | 10                      | 10                      | 35           |
 | spiral_print(matrix)                          | matrix/spiral_print.py                               | 12          | 10                      | 10                      | NLOC5        |
-| Function6                                     | Location6                                            | CCN6        | CCN6man1                | CCN6man2                | NLOC6        |
+| simulated_annealing(search)                   | searches/simulated_annealing.py                      | 16          | 15                | CCN6man2                | 79        |
```
Comment (mellonnen) [inline report.md]: I counted the 16 as well, so you can add that

File: complex_functions/SearchProblem.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
+class SearchProblem:
```
Comment (mellonnen) [inline complex_functions/SearchProblem.py:2]: We have had the convention to add the link to report on each file that we took from the repo, see the other complex functions as an example

File: complex_functions/simulated_annealing.py [hunk 3]
```diff
@@ -0,0 +1,137 @@
+# https://en.wikipedia.org/wiki/Simulated_annealing
```
Comment (mellonnen) [inline complex_functions/simulated_annealing.py:2]: See ⬆️

File: refactored_functions/SearchProblem.py [hunk 4]
```diff
@@ -0,0 +1,79 @@
+class SearchProblem:
```
Comment (mellonnen) [inline refactored_functions/SearchProblem.py:3]: Same here, see ⬆️

File: refactored_functions/simulated_annealing.py [hunk 5]
```diff
@@ -0,0 +1,167 @@
+# https://en.wikipedia.org/wiki/Simulated_annealing
```
Comment (mellonnen) [inline refactored_functions/simulated_annealing.py]: Same here

**Review by mellonnen — CHANGES_REQUESTED** [review]
Overall looks good! Just some minor comments. Should be a quick fix!

**Comment by PersonligaPersson** [comment]
All the above issues should now have been addressed.

**Pull request merged**